internal/locate/region_request.go

```diff
@@ -1931,6 +1931,11 @@ func (s *RegionRequestSender) onRegionError(
 	)
 
 	if s.replicaSelector != nil {
+		// the `mismatch peer id` error does not has a specific error type, so we have to match the error message.
```

Comment:
The above code line `metrics.TiKVRegionErrorCounter.WithLabelValues(regionErrorToLabel(regionErr)).Inc()` should be changed to so this error is not `unknown`.

Reply:
Changed.